File: AkkaUnitTest/src/main/java/org/akka/essentials/unittest/actors/SequencingActor.java
package org.akka.essentials.unittest.actors;

import java.util.List;

import akka.actor.ActorRef;
import akka.actor.UntypedActor;

public class SequencingActor extends UntypedActor {
	ActorRef next;
	List<Integer> head;
	List<Integer> tail;

	public SequencingActor(ActorRef next, List<Integer> head, List<Integer> tail) {
		this.next = next;
		this.head = head;
		this.tail = tail;
	}

	@Override
	public void onReceive(Object message) throws Exception {
		for (Integer value : head) {
			sender().tell(value);
		}
		sender().tell(message);
		for (Integer value : tail) {
			sender().tell(value);
		}
	}
}

File: AkkaUnitTest/src/main/scala/org/akka/essentials/unittest/example/SequencingActor.scala
package org.akka.essentials.unittest.example
import akka.actor.Actor
import akka.actor.ActorRef

class SequencingActor(next: ActorRef, head: List[Integer], tail: List[Integer]) extends Actor {

  def receive: Receive = {
    case message =>
      head map (next ! _)
      next ! message
      tail map (next ! _)
  }
}